### ISSUE TITLE
Prevent Tags From Being Created With Invalid Names

### ISSUE THREAD

**Issue by mstruve** [thepracticaldev/dev.to#5081]
**Describe the bug**
We are creating tags that have invalid names based on our tag name validation criteria
```ruby
validates :name, format: /\A[A-Za-z0-9\s]+\z/, allow_nil: true
```

There are currently 2087 tags in our system that have invalid names and the most recent one was created today. These invalid names contain special characters that then cause our `cached_with_tags` scope
```ruby
 scope :cached_tagged_with, ->(tag) { where("cached_tag_list ~* ?", "^#{tag},| #{tag},|, #{tag}$|^#{tag}$") }
```
to break like this:
```
ActionView::Template::Error: PG::InvalidRegularExpression: ERROR:  invalid regular expression: quantifier operand invalid
: SELECT COUNT(*) FROM "articles" WHERE "articles"."published" = TRUE AND (cached_tag_list ~* '^c++11,| c++11,|, c++11$|
```

**To Reproduce**
I am honestly not sure how invalid tags are being created since I have yet to be able to create one myself but there are recently created tags in our database that are invalid so it is happening somewhere

**Expected behavior**
Any failure to validate a tag should result in an error being raised to the user and the tag NOT being saved.

**Comment by bolariin** [comment]
I think the invalid tags might have been made using the API. #4239 is an attempt to resolve this.

**Comment by lightalloy** [comment]
A related issue that could be fixed along with this one #2934

**Comment by mstruve** [comment]
I think we can close this now. I just checked and since my cleanup and @maestromac's PR, there have been no new invalid name tags.
